Source/WebCore/page/Navigation.cpp

```diff
@@ -386,6 +386,8 @@ bool Navigation::hasEntriesAndEventsDisabled() const
 {
     if (!window()->document() || !window()->document()->isFullyActive())
         return true;
+    if (frame() && frame()->loader().isInitialAboutBlank())
+        return true;
```

Comment:
Do we really need another flag here? Can we get the same result by getting to the document loader and checking `isInFinishedLoadingOfEmptyDocument`?

Reply:
Unfortunately isInFinishedLoadingOfEmptyDocument has the same problem as the state machine, this is not static and in fact resets when the end of maybeLoadEmpty is reached. It will be even more important to keep the value static (until next navigation) once we schedule Navigate event dispatching on the event loop, like https://github.com/WebKit/WebKit/pull/28101 intends to do.